### PR TITLE
add license module

### LINKED ISSUE #7102: License module
It's a perfect module to start to contribute to this project

This module should allow to add:
- license MIT
- license Apache
- etc

### PR DESCRIPTION
**Pull request by cvirieux**
Fix #7102

### CODE REVIEW

**Comment by pascalgrimaud** [comment]
The CI failed, probably because of the merge of your other PR. You need to rebase and fix conflict

**Review by murdos — CHANGES_REQUESTED** [review]
I think it would be simpler to have only one module that would build multiple `JHipsterModule`, as done for e.g. by `CucumberAuthenticationModuleConfiguration`

File: src/main/resources/generator/setup/license/MIT.txt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+Copyright [YEAR] [COPYRIGHT HOLDER]
```
Comment (murdos) [inline src/main/resources/generator/setup/license/MIT.txt:1]: I think we could generate with the current year during generation

Comment (cvirieux) [reply src/main/resources/generator/setup/license/MIT.txt]: I have updated my code to add current year directly in the factory.

This is my first time working with mustache file so I don't know if it's the correct way to do it. Let me know if there is a better way

File: src/main/java/tech/jhipster/lite/generator/setup/license/domain/LicenseModuleFactory.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+public class LicenseModuleFactory {
+
+  public JHipsterModule buildMitModule(JHipsterModuleProperties properties) {
+    //@formatter:off
+    return moduleBuilder(properties)
+      .context()
+        .put("currentYear", Year.now().getValue())
+        .and()
+      .files()
+        .add(from("setup/license/MIT.txt.mustache"), to("./LICENSE.txt"))
```
Comment (murdos) [inline src/main/java/tech/jhipster/lite/generator/setup/license/domain/LicenseModuleFactory.java:18]: You can create a constant `SOURCE`:
`private static final JHipsterSource SOURCE = from("setup").append("license");`

And then use it here with the `template(String)` method that allows you to omit the `.mustache` suffix:
```suggestion
        .add(SOURCE.template("MIT.txt"), to("./LICENSE.txt"))
```

File: src/main/java/tech/jhipster/lite/generator/setup/license/domain/LicenseModuleFactory.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        .and()
+      .build();
+    //@formatter:on
+  }
+
+  public JHipsterModule buildApacheModule(JHipsterModuleProperties properties) {
+    //@formatter:off
+    return moduleBuilder(properties)
+      .files()
+      .add(from("setup/license/Apache.txt"), to("./LICENSE.txt"))
```
Comment (murdos) [inline src/main/java/tech/jhipster/lite/generator/setup/license/domain/LicenseModuleFactory.java:28]: Following previous comment suggesting to introduce a `SOURCE` constant:

```suggestion
      .add(SOURCE.append("Apache.txt"), to("./LICENSE.txt"))
```

File: src/main/java/tech/jhipster/lite/generator/setup/license/domain/LicenseModuleFactory.java (hunk 2 above)
Comment (murdos) [reply src/main/java/tech/jhipster/lite/generator/setup/license/domain/LicenseModuleFactory.java:18]: Usually we omit `./` (same for Apache):
```suggestion
        .add(from("setup/license/MIT.txt.mustache"), to("LICENSE.txt"))
```

File: src/main/resources/generator/setup/license/MIT.txt (hunk 1 above)
Comment (murdos) [reply src/main/resources/generator/setup/license/MIT.txt]: It looks fine!
I've added a few more suggestions, and then it'll be ok for me :-)

File: src/main/java/tech/jhipster/lite/generator/setup/license/infrastructure/primary/LicenseModuleConfiguration.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+@Configuration
+public class LicenseModuleConfiguration {
+
+  @Bean
+  JHipsterModuleResource mitLicenseModule(LicenseModuleService license) {
+    return JHipsterModuleResource
+      .builder()
+      .slug(MIT_LICENSE)
+      .withoutProperties()
+      .apiDoc("MIT license", "Add MIT license file")
```
Comment (murdos) [inline src/main/java/tech/jhipster/lite/generator/setup/license/infrastructure/primary/LicenseModuleConfiguration.java:22]: The fist argument is the group, so I think it should be the same for all module resources. 
```suggestion
      .apiDoc("License", "Add MIT license file")
```

**Review by murdos — DISMISSED** [review]

**Comment by murdos** [comment]
> Quick question : do you want me to squash all my commits or this is done during the merge ?

We usually don't squash during merge, so it's up to you :-)

**Pull request merged**